cmd/falco-exporter/root.go

```diff
@@ -27,21 +30,46 @@ func main() {
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 	pflag.Parse()
 
-	c, err := client.NewForConfig(config)
-	if err != nil {
-		log.Fatalf("unable to connect: %v", err)
+	var (
+		c   *client.Client
+		err error
+	)
+	exp := backoff.NewExponential(1*time.Second, 2*time.Minute)
+
+	// XXX(fujin): refactor into backoff() async func/closure or similar
+	for {
+		c, err = client.NewForConfig(config)
+		if err != nil {
+			log.Printf("unable to connect: %v, retrying after backoff of %d second(s)", err, exp.NextDuration/time.Second)
+			exp.Backoff()
+		} else {
+			exp.Reset()
+			break
+		}
 	}
 	defer c.Close()
-	outputClient, err := c.Output()
-	if err != nil {
-		log.Fatalf("unable to obtain an output client: %v", err)
+
+	var outputClient output.ServiceClient
+	for {
+		outputClient, err = c.Output()
+		if err != nil {
+			log.Printf("unable to obtain an output client: %v, retrying after backoff of %d second(s)", err, exp.NextDuration/time.Second)
+			exp.Backoff()
+		} else {
+			exp.Reset()
+			break
+		}
```

Comment:
[func Output() of client-go](https://github.com/falcosecurity/client-go/blob/3cb755d7594dc871f411b33b58a9366887bb4382/pkg/client/client.go#L75) checks for the connection then [just instantiates an object](https://github.com/falcosecurity/client-go/blob/3cb755d7594dc871f411b33b58a9366887bb4382/pkg/client/client.go#L80).

So I don't see how calling `c.Output()` multiple times could change the outcome, thus exponential backoff should not be used here.

Is there any other reason why we need exponential backoff here? If not, I would remove that.

Reply:
Sounds good, will remove.